### ISSUE TITLE
mips 32 compile err

### ISSUE THREAD

**Issue by loongson-gz** [mono/mono#14803]
os: debian 8 

make[4]: Entering directory '/root/mono/mono/mono/mini'
  CCLD     libmonoboehm-2.0.la
/usr/bin/ld: ../../libgc/.libs/libmonogc.a(checksums.o): ABI is incompatible with that of the selected emulation
/usr/bin/ld: failed to merge target specific data of file ../../libgc/.libs/libmonogc.a(checksums.o)
/usr/bin/ld: ../../libgc/.libs/libmonogc.a(real_malloc.o): ABI is incompatible with that of the selected emulation
/usr/bin/ld: failed to merge target specific data of file ../../libgc/.libs/libmonogc.a(real_malloc.o)
/usr/bin/ld: ../../libgc/.libs/libmonogc.a(backgraph.o): ABI is incompatible with that of the selected emulation
/usr/bin/ld: failed to merge target specific data of file ../../libgc/.libs/libmonogc.a(backgraph.o)
collect2: error: ld returned 1 exit status

**Comment by directhex** [comment]
big or little endian?

**Comment by loongson-gz** [comment]
little endian and the host cpu is loongson 3A3000

**Comment by directhex** [comment]
Is the build definitely using 32-bit throughout? There's a few places where we don't autodetect too well with 32-bit-userland-64-bit-kernel. If in doubt, stick a `linux32` in front of autogen and make

**Comment by loongson-gz** [comment]
yeah,  os and kenerl are 64bit , I throught debootstrap get  a debian 8 (mipsel)  os . then chroot to debian8 and make

**Comment by loongson-gz** [comment]
I stick a linux32 in front of autogen.sh and make. then  get the same err msg
